lib/WalletBackend.ts

```diff
@@ -2829,7 +2846,7 @@ export class WalletBackend extends EventEmitter {
             }
 
             if (this.shouldPerformAutoOptimize && this.autoOptimize) {
-                this.performAutoOptimize();
+                await this.performAutoOptimize();
```

Comment:
```suggestion
                this.performAutoOptimize();
```

This should remain non async. We don't want it to interrupt the periodic daemon updates.

Reply:
Added new event `autoOptimizeError` that at least catches the error and emits it, how about that?